cloudpickle/cloudpickle.py

```diff
@@ -1090,7 +1094,17 @@ def _make_skel_func(code, cell_count, base_globals=None):
     if base_globals is None:
         base_globals = {}
     elif isinstance(base_globals, str):
-        base_globals = vars(sys.modules[base_globals])
+        base_globals_name = base_globals
+        if base_globals_name in sys.modules.keys():
+            # this checks if we can import the previous environment the object
+            # lived in
+            base_globals = vars(sys.modules[base_globals_name])
+        elif base_globals_name in _dynamic_modules.keys():
```

Comment:
You could use the `get` function to simplify the code:

```python
base_globals = _dynamic_modules_globals.get(base_globals_name], {})
_dynamic_modules_globals[base_globals_name] = base_globals
```

I would also rename `_dynamic_modules` to `_dynamic_module_globals`